### PR TITLE
Fixes nREPL initialization with `:ack` port at cmd line

### PR DESCRIPTION
**Pull request by pfeodrippe**
Start nREPL with a ack port from the command line
raises a "java.util.IllegalFormatConversionException"
exception because the `format` expects a integer.

This fixes it by converting to int if `:ack` field
is given.

Before submitting a PR make sure the following things have been done:

- [x] The commits are consistent with our [contribution guidelines](CONTRIBUTING.md)
- [ ] You've added tests to cover your change(s)
- [ ] All tests are passing
- [x] The new code is not generating reflection warnings
- [x] You've updated the [changelog](../CHANGELOG.md)(that only applies to user-visible changes)

Thanks!

### CODE REVIEW

File: src/clojure/nrepl/cmdline.clj [hunk 1]
```diff
@@ -269,7 +269,7 @@
             greeting-fn (if (= transport #'transport/tty) #'transport/tty-greeting)
             server (start-server :port port :bind bind :handler handler
                                  :transport-fn transport :greeting-fn greeting-fn)]
-        (when-let [ack-port (:ack options)]
+        (when-let [ack-port (some-> (:ack options) ->int)]
```
Comment (bbatsov) [inline src/clojure/nrepl/cmdline.clj]: I guess this function is safe to call on a real integer, right?

Comment (pfeodrippe) [reply src/clojure/nrepl/cmdline.clj]: Yes, it checks if it's a number before parsing, it's used for `:port` param

Comment (bbatsov) [reply src/clojure/nrepl/cmdline.clj]: 👍

**Comment by bbatsov** [comment]
Thanks for tackling this!

**Comment by bbatsov** [comment]
One more argument for some cmdline tests I guess. 😉

**Comment by pfeodrippe** [comment]
Thanks for merging with the master o/

**Pull request merged**